pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderBasic.java

```diff
@@ -48,14 +51,28 @@ public void close() throws IOException {
 
     @Override
     public void initialize(ServiceConfiguration config) throws IOException {
-        File confFile = new File(System.getProperty(CONF_SYSTEM_PROPERTY_KEY));
-        if (!confFile.exists()) {
-            throw new IOException("The password auth conf file does not exist");
-        } else if (!confFile.isFile()) {
-            throw new IOException("The path is not a file");
+        String data = config.getProperties().getProperty(CONF_PULSAR_PROPERTY_KEY);
+        if (StringUtils.isEmpty(data)) {
+            data = System.getProperty(CONF_SYSTEM_PROPERTY_KEY);
+        }
+        if (StringUtils.isEmpty(data)) {
+            throw new IOException("No basic authentication config provided");
+        }
+
+        @Cleanup BufferedReader reader = null;
+        if (org.apache.commons.codec.binary.Base64.isBase64(data)) {
+            reader = new BufferedReader(new StringReader(new String(Base64.getDecoder().decode(data),
+                    StandardCharsets.UTF_8)));
+        } else {
```

Comment:
I think we can use 2 different config/property names instead of checking isBase64, the isBase64(path) maybe return true?

Reply:
I think we can add a prefix so like:
```
basicAuthConf=data:;base64,YOUR-BASE64
basicAuthConf=file:///my/basic.conf
``

Reply:
Ok, looks good to me.

Reply:
@nodece Could you please update the PR as your description?

Reply:
@codelipenghui This PR has been merged into master, this config still keep old style.

Reply:
Let me open a new PR to do that.

Reply:
Link to #16935.